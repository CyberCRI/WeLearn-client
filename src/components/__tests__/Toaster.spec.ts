import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import ToasterComponentVue from '../ToasterComponent.vue';

describe('ToasterComponent', () => {
  it('renders properly', () => {
    const wrapper = mount(ToasterComponentVue, {
      propsData: {
        description: 'Oops something wrong hapened. Please try again later.',
        level: 'error',
        title: 'Error'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toContain('Oops something wrong hapened. Please try again later.');
  });
});
