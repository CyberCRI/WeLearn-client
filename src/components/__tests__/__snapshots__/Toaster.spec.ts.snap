// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToasterComponent > renders properly 1`] = `
<div
  class="toaster error"
  data-v-ca54f05c=""
>
  <h1
    class="header"
    data-v-ca54f05c=""
  >
    Error
  </h1>
  <p
    class="body"
    data-v-ca54f05c=""
  >
    Oops something wrong hapened. Please try again later.
  </p>
</div>
`;
